Stratis.Bitcoin/Features/MemoryPool/MempoolErrors.cs

```diff
@@ -3,92 +3,214 @@
 
 namespace Stratis.Bitcoin.Features.MemoryPool
 {
+    /// <summary>
+    /// Exception thrown when memory pool exception occurs.
+    /// </summary>
     public class MempoolErrorException : Exception
     {
+        /// <summary>
+        /// Constructs a memory pool exception object.
+        /// Exception message is set from <see cref="MempoolValidationState.ErrorMessage"/>
+        /// </summary>
+        /// <param name="state">Validation state of the memory pool.</param>
         public MempoolErrorException(MempoolValidationState state) : base(state.ErrorMessage)
         {
             this.ValidationState = state;
         }
 
+        /// <summary>Gets the validation state of the memory pool.</summary>
         public MempoolValidationState ValidationState
         {
             get;
             private set;
         }
```

Comment:
as per new code style, we use one liners for this

`public MempoolValidationState ValidationState { get; private set; }`

this one already is accepted, but see https://github.com/stratisproject/StratisBitcoinFullNode/issues/258